src/main/kotlin/it/pagopa/wallet/scheduler/jobs/paymentwallet/OnboardedPaymentWalletJob.kt

```diff
@@ -24,82 +28,40 @@ import reactor.core.publisher.Mono
 @Component
 class OnboardedPaymentWalletJob(
     @Autowired private val walletService: WalletService,
-    @Autowired private val cdcEventDispatcherService: CdcEventDispatcherService
+    @Autowired private val cdcEventDispatcherService: CdcEventDispatcherService,
+    @Autowired private val redisResumePolicyService: RedisResumePolicyService
 ) : ScheduledJob<OnboardedPaymentWalletJobConfiguration, String> {
 
     private val logger = LoggerFactory.getLogger(javaClass)
     override fun id(): String = "onboarded-payment-wallet-job"
 
     override fun process(configuration: OnboardedPaymentWalletJobConfiguration): Mono<String> {
-        /*
-           TODO: add redis checkpoint logic here to retrieve the latest processed wallet
-           recover from redis the latest sent event date
-           if null -> use start date
-           else -> use valued last created wallet processed date (sliding window)
-        */
         val startDate = configuration.startDate
         val endDate = configuration.endDate
         logger.info("Starting payment wallet processing in time window {} - {}", startDate, endDate)
-        return walletService
-            .getWalletsForCdcIngestion(startDate = startDate, endDate = endDate)
+        return Mono.fromCallable { redisResumePolicyService.getResumeTimestamp(id()) }
+            .subscribeOn(Schedulers.boundedElastic())
+            .flatMapMany {
```

Comment:
can it make sense to have some logging here
`logger.info("Fetching wallets for CDC ingestion between {} and {}", startDate, endDate)` 
 for better traceability and debugging